gen/ldap_it4i

```diff
@@ -0,0 +1,105 @@
+our $A_SSHKEYS;  *A_SSHKEYS = \'urn:perun:user:attribute-def:def:sshPublicKey';
+our $A_MEMBER_STATUS;  *A_MEMBER_STATUS =  \'urn:perun:member:attribute-def:core:status';
+
+our $STATUS_VALID;      *STATUS_VALID =        \'VALID';
+our $STATUS_EXPIRED;    *STATUS_EXPIRED =      \'EXPIRED';
+our $STATUS_INVALID;    *STATUS_INVALID =      \'INVALID';
+our $STATUS_DISABLED;   *STATUS_DISABLED =     \'DISABLED';
+
+# check facility attribute
+my $ldapBaseDN = $data->getFacilityAttributeValue( attrName => $A_F_BASE_DN );
+if (!defined($ldapBaseDN)) {
+	exit 1;
+}
+
+# gather user data
+my $users;
+
+foreach my $resourceId ( $data->getResourceIds() ) {
+	foreach my $memberId ( $data->getMemberIdsForResource( resource => $resourceId )) {
+
+		my $login = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LOGIN_EINFRA );
+		$users->{$login}->{$A_FIRST_NAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_FIRST_NAME );
+		$users->{$login}->{$A_LAST_NAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_LAST_NAME );
+		$users->{$login}->{$A_DISPLAY_NAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_DISPLAY_NAME );
+		$users->{$login}->{$A_USER_PREFERRED_MAIL} = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_PREFERRED_MAIL );
+		$users->{$login}->{$A_SSHKEYS} = $data->getUserAttributeValue( member => $memberId, attrName => $A_SSHKEYS );
+
+		my $status = $data->getMemberAttributeValue( member => $memberId, attrName => $A_MEMBER_STATUS );
+		if ($status eq $STATUS_VALID) {
+			$status = 'ACTIVE';
+		} elsif ($status eq $STATUS_INVALID) {
+			$status = 'INCOMPLETE';
+		} elsif ($status eq $STATUS_EXPIRED) {
+			$status = 'INACTIVE';
+		} elsif ($status eq $STATUS_DISABLED) {
+			$status = 'ARCHIVED';
+		}
+		$users->{$login}->{$A_MEMBER_STATUS} = $status;
+	}
+}
+
+# print BASE_DN file
+open FILE,">:encoding(UTF-8)","$baseDnFileName" or die "Cannot open $baseDnFileName: $! \n";
+print FILE $ldapBaseDN;
+close(FILE);
+
+# print user data LDIF
+open FILE,">:encoding(UTF-8)","$fileName" or die "Cannot open $fileName: $! \n";
+
+for my $login (sort keys %$users) {
+
+	print FILE "dn: uid=" . $login . "," . $ldapBaseDN . "\n";
+	print FILE "uid: " . $login . "\n";
+	print FILE "givenName: " . $users->{$login}->{$A_FIRST_NAME} . "\n";
+	print FILE "sn: " . $users->{$login}->{$A_LAST_NAME} . "\n";
+	print FILE "fullName: " . $users->{$login}->{$A_FIRST_NAME} . " " . $users->{$login}->{$A_LAST_NAME} . "\n";
+	print FILE "displayName: " . $users->{$login}->{$A_DISPLAY_NAME} . "\n";
+	print FILE "mail: " . $users->{$login}->{$A_USER_PREFERRED_MAIL} . "\n";
+	print FILE "userPassword: {SASL}" . $login . '@EINFRA' . "\n";
+	print FILE "state: " . $users->{$login}->{$A_MEMBER_STATUS} . "\n";
+
+	my $sshKeys = $users->{$login}->{$A_SSHKEYS};
+	foreach my $sshKey (@$sshKeys) {
```

Comment:
I think simple retyping to array will fail if `$sshKeys` is `undef`. Maybe add check for it (wrap cycle in IF). Also you can sort ssh keys themselves, so they don't differ between two script runs.

Reply:
I have tested this and even for `undef` value, the cycle just does not go through any values and does not fail, so I do not think the check is necessary. Also there is a function in ADConnector for comparing unsorted values used in most of the ad scripts, I have just switch to this instead of plain `~~`, I think it is better than sorting the values in gen script. Or do you think I should sort them anyway?

Reply:
Then it should be OK, regarding the `undef`. Sorting in gen script is necessary when we want to compare gen outputs (eg. when changing the gen script).

Reply:
Good point, I've added the sort and in that case, the check is needed.